tmt/steps/execute/internal.py

```diff
@@ -192,7 +319,39 @@ def log(key, value=None, color=None, shift=1, level=1):
                 command, cwd=workdir, env=environment,
                 join=True, interactive=self.get('interactive'), log=log,
                 timeout=tmt.utils.duration_to_seconds(test.duration))
-            test.returncode = 0
+            # Check to see if the rstrnt-report-result script was used
+            # and has generated a report.
+            result_file_found = \
+                int("{0}".format(
+                    guest.execute("ls {0} | wc -l".format(
+                                  RSTRNT_REPORT_RESULT_OUTPUT)
+                                  )[0]
+                    )
+                    )
+            if result_file_found:
+                ret = \
+                    guest.execute("cat {0}".format(
```

Comment:
Wouldn't it make sense to store the file inside tmt's workdir, hence it would be pulled during `guest.pull()` and you could easily access it from here, rather than having to `cat` the file? Moreover, we ideally want to keep the `guest.execute()` calls at minimum since establishing ssh connection takes time and there were some complaints about tmt being slow :)

Reply:
That sounds logical. I've made that change.